### DISCUSSION TITLE
2.7.x/docs/web/senior-websocket

### DISCUSSION THREAD

**Discussion by giscus** [gogf/gf-site#107] [General]
# 2.7.x/docs/web/senior-websocket

使用GoFrame框架开发WebSocket服务，包括HTML5客户端、WebSocket服务端、HTTPS支持和安全校验等内容。通过示例代码，展示了WebSocket的连接与信息传递机制，并详细说明了服务端与客户端之间的通信流程和数据处理方法，让读者能够快速上手并实现功能齐全的WebSocket服务。

https://goframe.org/2.7.x/docs/web/senior-websocket

**Comment by BuxsRen** [comment]
基于GoFrame 2.7开发的WebSocket示例，包含客户端和服务端

已实现的功能

仿http路由与路由组写法
路由支持链式调用与路由组层级嵌套
路由/组与中间件实现
通过中间件可实现消息限流(选)
Action绑定到控制器
群聊与私聊，上下线通知
支持单点登录被迫下线(选)
异常处理，隐式ping，客户端自动重连等
仓库代码
https://github.com/BuxsRen/cloud-clipboard
